tests/__init__.py

```diff
@@ -359,3 +362,66 @@ def assert_url_equal(url1, url2):
     assert_equal(parts1.hostname, parts2.hostname)
     assert_equal(parts1.port, parts2.port)
     assert_equal(parse_qs(parts1.query), parse_qs(parts2.query))
+
+
+class RawResponse(BytesIO):
+    def stream(self, **kwargs):
+        contents = self.read()
+        while contents:
+            yield contents
+            contents = self.read()
+
+
+class BotocoreHTTPStubber(object):
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self._requests = []
+        self._responses = []
+
+    def add_response(self, response):
+        self._responses.append(response)
+
+    def create_response(self, url=None, status=200, headers=None, body=None):
```

Comment:
What's your thoughts on just having a single `add_response()` method? It would optionally take the `url`, `status`, `headers`, `body`, and `response` (the `response` would be mutually exclusive with the `url`, `status`, etc.). Then we would also expose an `add_error()` method that accepts an exception. The nice part about this is that it more closely follows the `Stubber` interface and it is very explicit how to provide responses/exceptions.